.github/workflows/build-publish-linux.yml

```diff
@@ -0,0 +1,49 @@
+name: Build and publish linux wheel
+
+on:
+  push:
+    branches: publish-linux
+
+permissions:
+  contents: read # to fetch code (actions/checkout)
+
+jobs:
+  build:
+    runs-on: ubuntu-latest  # todo, add more OS systems to see if they work as well
+    strategy:
+      matrix:
+        python-version: ['3.7', '3.8', '3.9', '3.10']  # '3.11'
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install system packages
+        run: |
+          sudo apt-get update
+          sudo apt-get install git cmake capnproto zlib1g-dev g++ build-essential python3-dev python3-pip pkg-config libzip-dev software-properties-common libbz2-dev
+
+      - name: Install pip packages
+        run: |
+          python -m pip install --upgrade pip cmake wheel setuptools
+
+      - name: Build wheel
+        run: |
+          python_header_path=$(python3-config --cflags | awk -F '-I' '{print $2}' | awk '{print $1}')
+          cmake . -DBUILD_TESTS=ON -DBUILD_MANYLINUX=ON -DPYTHON_INCLUDE_DIR=$python_header_path
+          python setup.py -q build_ext -i -j3
+          make -j3
```

Comment:
By default, retro builds with `testing` On which means that GZip is included in the build process. 
I haven't looked at the other options but we should check which options are being used to build with